### PR TITLE
[MM-49086] Implement authentication check and re-registration logic in case of failure

### PR DESCRIPTION
**Pull request by streamer45**
#### Summary

PR adds some logic to:

- Login to the job service on plugin activation. This makes sure that everything is configured correctly past the initial registration.
- Attempt to re-register if the first login attempt fails. This is particularly important to prevent getting locked out and requiring resetting some state in the k/v store.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-49086

### CODE REVIEW

File: server/job_service.go
Comment (cpoile) [inline server/job_service.go:185]: Just curious why we use a cluster mutex here, and not above?

**Review by cpoile — APPROVED** [review]
Approving assuming you know what you're doing, but I have one question. :)

File: server/job_service.go
Comment (streamer45) [reply server/job_service.go:185]: It's used above as well, just buried in `getJobServiceClientConfig`

Comment (cpoile) [reply server/job_service.go:185]: Ah. Would it be too much to consolidate both of these, so one of them doesn't get accidentally removed or whatnot? Up to you.

Comment (streamer45) [reply server/job_service.go:185]: Maybe we could call it just once at the beginning. The down side is that all other instances will be blocked on this for potentially some time as there are requests being made but I guess during init that would be acceptable. I'll take another look.

Comment (streamer45) [reply server/job_service.go]: Thanks @cpoile , looks cleaner now and works as expected :+1:

**Pull request merged**
